arkouda/client.py

```diff
@@ -49,9 +50,36 @@
 logger = getArkoudaLogger(name="Arkouda Client")
 clientLogger = getArkoudaLogger(name="Arkouda User Logger", logFormat="%(message)s")
 
-# Print splash message
-print("{}".format(pyfiglet.figlet_format("Arkouda")))
-print(f"Client Version: {__version__}")  # type: ignore
+
+class ClientMode(Enum):
+    """
+    The ClientMode enum provides controlled vocabulary indicating whether the
+    Arkouda client is in UI mode or API mode. If in API mode, it is assumed the
+    Arkouda client is being used via an API call instead of a Python shell or notebook.
+    """
+    UI = 'UI'
+    API = 'API'
+
+    def __str__(self) -> str:
+        """
+        Overridden method returns value.
+        """
+        return self.value
+
+    def __repr__(self) -> str:
+        """
+        Overridden method returns value.
+        """
+        return self.value
+
+
+# Get ClientMode, defaulting to UI
+mode = ClientMode(os.getenv('ARKOUDA_CLIENT_MODE', 'UI'))
```

Comment:
would it be a good idea to change this to
```python
mode = ClientMode(os.getenv('ARKOUDA_CLIENT_MODE', 'UI').upper())
```
or something like that so it case doesn't matter when setting the env var? it prob doesn't matter too much

Reply:
@pierce314159 good points on both, will implement

Reply:
@pierce314159 done :)